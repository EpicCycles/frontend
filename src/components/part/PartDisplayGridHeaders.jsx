import React from 'react';
import * as PropTypes from 'prop-types';
import PartHeaders from './PartHeaders';
import SupplierProductFieldHeaders from '../supplierProduct/SupplierProductFieldHeaders';
import { fixedHeaderClassname } from '../app/model/helpers/display';

const PartDisplayGridHeaders = props => {
  const { className, lockFirstColumn, showErrors, showSupplierProducts, includeActions } = props;
  return (
    <div className="grid-row grid-row--header " key="part-display-grid-header-row">
      <div
        className={`grid-item--header ${className} ${fixedHeaderClassname(lockFirstColumn)}`}
        key="part-display-grid-header-section"
      >
        Section
      </div>
      <PartHeaders
        showErrors={showErrors}
        className={className}
        key="part-display-grid-header-part-headers"
      />
      {showSupplierProducts && (
        <SupplierProductFieldHeaders
          className={className}
          showErrors={showErrors}
          key="part-display-grid-header-sp-headers"
        />
      )}
      {includeActions && (
        <div className={`grid-item--header ${className}`} key="part-display-grid-header-actions">
          Actions
        </div>
      )}
    </div>
  );
};
PartDisplayGridHeaders.defaultProps = {
  className: '',
};
PartDisplayGridHeaders.propTypes = {
  lockFirstColumn: PropTypes.bool,
  showSupplierProducts: PropTypes.bool,
  showErrors: PropTypes.bool,
  includeActions: PropTypes.any,
  className: PropTypes.string,
};
export default PartDisplayGridHeaders;
