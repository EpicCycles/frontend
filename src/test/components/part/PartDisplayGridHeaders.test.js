import React from 'react';
import toJson from 'enzyme-to-json';
import PartDisplayGridHeaders from '../../../components/part/PartDisplayGridHeaders';

test('should display just part headers when supplier products no requested', () => {
  const component = shallow(<PartDisplayGridHeaders />);
  expect(toJson(component)).toMatchSnapshot();
});
test('should display all headers with locking and an extra class', () => {
  const component = shallow(
    <PartDisplayGridHeaders lockFirstColumn showSupplierProducts showErrors className="pink" />,
  );
  expect(toJson(component)).toMatchSnapshot();
});
test('should display all headers with locking and an extra class and actions', () => {
  const component = shallow(
    <PartDisplayGridHeaders
      lockFirstColumn
      showSupplierProducts
      showErrors
      className="pink"
      includeActions
    />,
  );
  expect(toJson(component)).toMatchSnapshot();
});
