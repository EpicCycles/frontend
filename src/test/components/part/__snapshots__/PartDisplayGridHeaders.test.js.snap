// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display all headers with locking and an extra class 1`] = `
<div
  className="grid-row grid-row--header "
  key="part-display-grid-header-row"
>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="part-display-grid-header-section"
  >
    Section
  </div>
  <PartHeaders
    className="pink"
    key="part-display-grid-header-part-headers"
    lockFirstColumn={false}
    showErrors={true}
  />
  <SupplierProductFieldHeaders
    className="pink"
    key="part-display-grid-header-sp-headers"
    lockFirstColumn={false}
    showErrors={true}
  />
</div>
`;

exports[`should display all headers with locking and an extra class and actions 1`] = `
<div
  className="grid-row grid-row--header "
  key="part-display-grid-header-row"
>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="part-display-grid-header-section"
  >
    Section
  </div>
  <PartHeaders
    className="pink"
    key="part-display-grid-header-part-headers"
    lockFirstColumn={false}
    showErrors={true}
  />
  <SupplierProductFieldHeaders
    className="pink"
    key="part-display-grid-header-sp-headers"
    lockFirstColumn={false}
    showErrors={true}
  />
  <div
    className="grid-item--header pink"
    key="part-display-grid-header-actions"
  >
    Actions
  </div>
</div>
`;

exports[`should display just part headers when supplier products no requested 1`] = `
<div
  className="grid-row grid-row--header "
  key="part-display-grid-header-row"
>
  <div
    className="grid-item--header  "
    key="part-display-grid-header-section"
  >
    Section
  </div>
  <PartHeaders
    className=""
    key="part-display-grid-header-part-headers"
    lockFirstColumn={false}
    showErrors={false}
  />
</div>
`;
